internal/storage/sql/sqlite/sqlite.go

```diff
@@ -151,6 +151,9 @@ func (s *Store) CreateRollout(ctx context.Context, r *flipt.CreateRolloutRequest
 		var serr sqlite3.Error
 
 		if errors.As(err, &serr) && serr.Code == sqlite3.ErrConstraint {
+			if segment := r.GetSegment(); segment != nil {
+				return nil, errs.ErrNotFoundf(`flag "%s/%s or segment %s"`, r.NamespaceKey, r.FlagKey, segment.SegmentKey)
+			}
```

Comment:
Same as the above, but I have a sneaking suspicion it might be event harder with SQLite errors. Not sure there is as-much metadata:
https://github.com/mattn/go-sqlite3/blob/f08f1b6b9ce62b2496d8d64df26c1e278887bc1c/error.go#L27-L34